qt_ui/windows/mission/flight/QFlightCreator.py

```diff
@@ -255,3 +263,22 @@ def update_max_size(self, available: int) -> None:
 
         default_size = max(2, available, aircraft.max_group_size)
         self.flight_size_spinner.setValue(default_size)
+
+        self.pilots_changed.emit()
+
+    def on_pilot_selected(self):
+        # Pilot selection detected. If this is a player flight, set start_type
+        # as configured for players in the settings.
+        # Otherwise, set the start_type as configured for AI.
+        # https://github.com/dcs-liberation/dcs_liberation/issues/1567
+
+        roster = self.roster_editor.roster
+
+        if roster is not None and roster.player_count > 0:
+            start_type = self.game.settings.default_start_type_client
+        else:
+            start_type = self.game.settings.default_start_type
+
+        for i, st in enumerate([b for b in ["Cold", "Warm", "Runway", "In Flight"]]):
+            if start_type.value == st:
+                self.start_type.setCurrentIndex(i)
```

Comment:
Perhaps `self.start_type.setCurrentText(start_type.value)` is a better approach? 😅

Reply:
Done 👍